metadata/src/main/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -328,8 +328,21 @@ public void register(MetaLogListener listener) throws Exception {
 
     @Override
     public long scheduleWrite(long epoch, List<ApiMessageAndVersion> batch) {
-        return shared.tryAppend(nodeId, leader.epoch(), new LocalRecordBatch(
-            batch.stream().map(r -> r.message()).collect(Collectors.toList())));
+        return scheduleAtomicWrite(epoch, batch);
+    }
+
+    @Override
+    public long scheduleAtomicWrite(long epoch, List<ApiMessageAndVersion> batch) {
+        return shared.tryAppend(
```

Comment:
nit: since we're just moving this code, can we avoid reformatting it?

Reply:
In general, I agree but can we keep this reformatting? I found the old formatting hard to read and technically we are adding a new method called `scheduleAtomicWrite`.

Reply:
Yea, sounds good. I actually agree it's more readable now